pkg/apis/intevents/v1alpha1/brokercell_types.go

```diff
@@ -71,11 +71,29 @@ var (
 // ComponentParameters specifies scaling and resource parameters to be used
 // by a single component of a BrokerCell.
 type ComponentParameters struct {
+	// AvgCPUUtilization specifies the average CPU consumption targeted by the component's Horizontal Pod Autoscaler
+	AvgCPUUtilization *int32 `json:"avgCPUUtilization,omitempty"`
+
+	// AvgMemoryUsage specifies the average memory consumption targeted by the component's Horizontal Pod Autoscaler
+	AvgMemoryUsage *string `json:"avgMemoryUsage,omitempty"`
+
+	// CPURequest specifies the minimal amount of the CPU for the deployment to be schedulable
+	CPURequest *string `json:"cpuRequest,omitempty"`
+
+	// CPULimit specifies the maximal amount of the CPU to be consumable by the deployment
+	CPULimit *string `json:"cpuLimit,omitempty"`
+
 	// MinReplicas specifies the minimum replica count for the component.
 	MinReplicas *int32 `json:"minReplicas,omitempty"`
 
 	// MaxReplicas specifies the maximum replica count for the component.
 	MaxReplicas *int32 `json:"maxReplicas,omitempty"`
+
```

Comment:
nit: Let's move these up above the MinReplicas and below the CPULImit

Reply:
Sure